### PR TITLE
feat: use updated clair-action/claircore

### PR DESCRIPTION
**Pull request by zregvart**
We have found[^1] that the Red Hat VEX[^2] data consistently contains the release date information, and the OVAL[^3] data consistently does not. Recently[^4] claircore introduced the Red Hat VEX updater and the clair-action was updated[^5] to use that version.

This updates to clair-action 0.0.9 which updates claircore to 1.5.30 containing the needed Red Hat VEX support.

Resolves: https://issues.redhat.com/browse/EC-835

[^1]: https://issues.redhat.com/browse/EC-812
[^2]: https://www.redhat.com/en/blog/red-hat-vex-files-cves-are-now-generally-available
[^3]: https://access.redhat.com/articles/221883/
[^4]: https://issues.redhat.com/browse/CLAIRDEV-28
[^5]: https://issues.redhat.com/browse/EC-854

### CODE REVIEW

**Comment by zregvart** [comment]
I'm putting this in draft until we figure out why we're missing some of the vulnerabilities. See https://github.com/quay/claircore/issues/1409

**Comment by zregvart** [comment]
I'm going to put this out of draft. I'd like to see clair-action 0.0.10 release with the update to claircore 1.5.31. [This change](https://github.com/quay/claircore/pull/1410) in particular would be useful for stability purposes, otherwise timeouts might occur when processing Red Hat VEX data. So the workaround here for now is to build and update the version of claircore to 1.5.31.

Other issues of note. With this version I can see different number of vulnerabilities reported for the same image. I'm using `quay.io/zregvart_redhat/vuln` as an example vulnerable image. It would be good if there were a set of images that we know the expected reported vulnerabilities so we can compare as we make changes here.

I've attached the condensed output of the two different versions (current and proposed here):

- [v0.0.8-report.txt](https://github.com/user-attachments/files/17043504/v0.0.8-report.txt) (current)
- [v0.0.9-plus-report.txt](https://github.com/user-attachments/files/17043506/v0.0.9-plus-report.txt) (proposed here)

I've used the following `jq` filter to generate those text files:

    jq -r '(["ISSUED", "SEVERITY", "NAME", "LINKS"] | (., map(length*"-"))), (.vulnerabilities[] | [.issued, .normalized_severity, .name, .links]) | @tsv'

The differences might be as expected, and it would be nice if they were double checked. For our use case in [EC-835](https://issues.redhat.com/browse/EC-835) I'm pretty happy that we now have the issue/release/public date present in the output.

There is room for improvement here, I think the clair-action and claircore tend to deal well with incremental updates, it might be a good idea to use the existing clair-in-ci image as a base for the second stage, when the database is updated, so it has access to the previous version of the database. I have not tried this.

**Review by dirgim — APPROVED** [review]
This looks good to me, but I'll defer to more knowledgeable members of the team for the final call

**Review by simonbaird — APPROVED** [review]

**Comment by lcarva** [comment]
Maybe @jencull can do a final review here?

**Review by jencull — APPROVED** [review]
LGTM

**Comment by lcarva** [comment]
Please let me know if there's anything else needed to have this merged :pray:

**Comment by jencull** [comment]
> Please let me know if there's anything else needed to have this merged 🙏

@lcarva @jsztuka might be the best person for a final once over, in case you need anything else before merging. Returns to work tomorrow, I believe. I will raise it with him.

**Pull request merged**
